### PR TITLE
feat: build sc4snmp with gopoller and add logger configuration in go code

### PR DESCRIPTION
**Pull request by wojtekzyla**
# Description

This PR updates the Dockerfile in order to build sc4snmp with a new poller written in GO. This PR also adds configuration of logger in the new GO poller.

## Type of change

Please delete options that are not relevant.
- [ ] New feature

## How Has This Been Tested?

Manual testing

## Checklist

- [ ] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] I have run pre-commit on all files before creating the PR
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp/gopoller/auth.go [hunk 1]
```diff
@@ -75,6 +77,10 @@ func getSecurityParameters(authData SnmpV3StringAuthData) (*gosnmp.UsmSecurityPa
 		PrivacyPassphrase:        authData.PrivacyPassphrase,
 		AuthoritativeEngineID:    authData.AuthoritativeEngineID,
 	}
+	useLogger := os.Getenv("USE_GO_LOGGER")
```
Comment (ajasnosz) [inline splunk_connect_for_snmp/gopoller/auth.go:80]: maybe we could name it `ENABLE_GO_LOGGER` or something similar? 
Is it possible to set log level for this logger?
Could we add setting this flag to yaml? I think we could do section for go under
`poller.go` or `worker.poller.go`

**Review by ajasnosz — CHANGES_REQUESTED** [review]

File: splunk_connect_for_snmp/gopoller/auth.go (hunk 1 above)
Comment (wojtekzyla) [reply splunk_connect_for_snmp/gopoller/auth.go:80]: I can change the name to `ENABLE_GO_LOGGER` and add the configuration to values.yaml. When it comes to the log level, I found the following info in the gosnmp documentation: 
Screenshot 2024-02-01 at 13 51 26

I also checked how this logger is used in code of gosnmp, and they don't distinguish any logging levels. They just simply log if the logger was configured

File: charts/splunk-connect-for-snmp/values.yaml [hunk 2]
```diff
@@ -226,6 +226,10 @@ poller:
   # https://splunk.github.io/splunk-connect-for-snmp/main/configuration/poller-configuration/#define-usernamesecrets
   usernameSecrets: []
 
+  go:
+    # Enable logs from gosnmp library
+    gosnmpLogger: false
```
Comment (ajasnosz) [inline charts/splunk-connect-for-snmp/values.yaml:231]: probably we should add it also here https://github.com/splunk/splunk-connect-for-snmp/blob/main/charts/splunk-connect-for-snmp/templates/worker/_helpers.tpl#L81 as env variable in workers, and map this param to ENABLE_GO_LOGGER and I think we can set the default as false.

**Review by ajasnosz — CHANGES_REQUESTED** [review]

File: splunk_connect_for_snmp/gopoller/auth.go (hunk 1 above)
Comment (ajasnosz) [reply splunk_connect_for_snmp/gopoller/auth.go]: Ok, so if it logs everything, we'll follow with that.

File: charts/splunk-connect-for-snmp/values.yaml (hunk 2 above)
Comment (wojtekzyla) [reply charts/splunk-connect-for-snmp/values.yaml]: Done

**Review by ajasnosz — APPROVED** [review]

**Pull request merged**